### PR TITLE
[CBRD-25077] Add Built-in package information in db_stored_procedure catalog

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-25077

- Add pkg_name, is_system_generated in _db_stored_procedure, _db_stored_procedure_args in system catalog module
- Add "WHERE is_system_genenrated = 0" in the system view queries related to stored procedures
- Refactor registring stored procedures not only from SQL query, but code routine.
  - Split jsp_cl into two parts, SQL query and system catalog interface.
- Refactor adding built-in procedures (`DBMS_OUTPUT`) routine
  - By using the system catalog inteface (sp_catalog.cpp), Add DBMS_OUTPUT built-in procedures is implemented in `boot_initialize_client ()` of boot_cl.c. see `sp_builtin_install ()` in sp_catalog.cpp.
  - the old implementations (in util_sa.c) is removed
- Check is_system_generated in dropping stored procedure routine
- In unloaddb, the system generated stored procedures are skipped by checking the `is_system_genenrated` column.

NOTE: This PR is based on [CBRD-24921](http://jira.cubrid.org/browse/CBRD-24921)

### CODE REVIEW

**Review by hyunikn — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

File: src/transaction/boot_cl.c [hunk 1]
```diff
@@ -565,6 +566,11 @@ boot_initialize_client (BOOT_CLIENT_CREDENTIAL * client_credential, BOOT_DB_PATH
 		  error_code = tran_commit (false);
 		}
 	    }
+
+	  if (error_code == NO_ERROR)
+	    {
+	      error_code = sp_builtin_install ();
+	    }
```
Comment (ctshim) [inline src/transaction/boot_cl.c]: Trivial, how about moving to line 567?

File: src/object/schema_system_catalog_install_query_spec.cpp [hunk 2]
```diff
@@ -1205,9 +1209,10 @@ sm_define_view_stored_procedure_arguments_spec (void)
 	"FROM "
 	  /* CT_STORED_PROC_ARGS_NAME */
 	  "[%s] AS [sp] "
+        "WHERE [sp].[is_system_generated] = 0 "
 	"ORDER BY " /* Is it possible to remove ORDER BY? */
 	  "[sp].[sp_name], "
-	  "[sp].[index_of]",
+	  "[sp].[index_of] ",
```
Comment (ctshim) [inline src/object/schema_system_catalog_install_query_spec.cpp]: Do you need some space?

File: src/jsp/sp_catalog.hpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+int sp_builtin_install ();
+
+// insert into system catalogs
+int sp_add_stored_procedure (SP_INFO &info);
+int sp_add_stored_procedure_argument (MOP *mop_p, SP_ARG_INFO &info);
+
+// misc
+void sp_normalize_name (std::string &s);
+
+#endif // _SP_DEFINITION_HPP_
```
Comment (ctshim) [inline src/jsp/sp_catalog.hpp]: It would be nice if you could add a newline character.

File: src/jsp/sp_catalog.cpp [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+
+  AU_ENABLE (save);
+  return err;
+}
+
+void sp_normalize_name (std::string &s)
+{
+  s.resize (SM_MAX_IDENTIFIER_LENGTH);
+  sm_downcase_name (s.data (), s.data (), SM_MAX_IDENTIFIER_LENGTH);
+}
```
Comment (ctshim) [inline src/jsp/sp_catalog.cpp]: It would be nice if you could add a newline character.

**Review by ctshim — APPROVED** [review]

**Pull request merged**